aesara/tensor/shape.py

```diff
@@ -411,12 +430,10 @@ def make_node(self, x, shape):
         if not isinstance(x, Variable):
             x = at.as_tensor_variable(x)
 
-        shape = at.as_tensor_variable(shape, ndim=1)
-
-        if isinstance(shape, Constant):
-            shape = tuple(shape.data)
-        else:
-            shape = tuple(at.as_tensor_variable(s, ndim=0) for s in shape)
+        shape = tuple(
+            at.as_tensor_variable(s, ndim=0)
+            for s in at.as_tensor_variable(shape, ndim=1)
+        )
```

Comment:
This will introduce unnecessary `Subtensor` operations when `shape` is a constant, and an increased need for constant folding.